cpp/src/slice2php/Main.cpp

```diff
@@ -271,12 +264,6 @@ CodeVisitor::visitClassDefStart(const ClassDefPtr& p)
 
     _out << eb; // End of class.
 
-    if (_classHistory.count(scoped) == 0 && p->canBeCyclic())
-    {
-        // Emit a forward declaration for the class in case a data member refers to this type.
-        _out << sp << nl << type << " = IcePHP_declareClass('" << scoped << "');";
-    }
```

Comment:
Impossible to hit.
`visitClassDecl` also checks for `_classHistory.count(scoped) == 0` and inserts `scoped` if so. Since we always call `visitClassDecl` before `visitClassDef`, by the time we get here, the count will _always_ be non-zero.